XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandMeshUpdatedEventData.cs

```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine;
+using XRTK.Definitions.Utilities;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    /// <summary>
+    /// See BaseHandVisualizer.OnHandMeshUpdated for an example of how to use the
+    /// hand mesh info to render a mesh.
+    /// </summary>
+    public class HandMeshUpdatedEventData
```

Comment:
Seems like a lot of data to be passing around. Why are we doing this?

Can't we just handle all the data inside the hand data provider?

Reply:
No you don't want the hand provider to be in charge of rendering the hand.

Say you have another platform that produces hand mesh data like this - you'd want to be able to reuse a hand rendering component, especially if you were tempted to add physics.

I'd you keep it locked in the hand data provider, devs won't be able to make use of the data without modifying it at the platform implementation level.

Reply:
Agreed, 
* DataProviders produce Data from a platform in a consistent / interoperable way.  
* Visualisers have the job of rendering 
* Mappings allow the user to change the actions the outputs of a "Hand controller" do

Reply:
The dataprovider point is a significant departure from how the MRTK is handling this.
Would you be so kind as to draw up a relationship diagram explaining how the data provider can be accessed by apps?

Reply:
Ok I understand more. Its just a part of the controller. 

So you get the controller, but then you have to cast it into a hand type, before querying it for its joints. I guess that works, but it leaves a lot of boilerplate for accessing hand joint data.